### PR TITLE
ai_model.py

### PR DESCRIPTION
**Pull request by Horlabrainmoore**
from sklearn.ensemble import RandomForestClassifier import numpy as np
import joblib

# Training Data (Example dataset)
X_train = np.array([
    [0.5, 10000, 1],  # Legitimate Transaction
    [1.2, 200000, 0], # High-risk CoinJoin
    [0.7, 50000, 1],  # Medium Risk
    [2.5, 100, 0],    # Micro-payment spam
])
y_train = np.array([0, 1, 0, 1])  # Labels: 0 = Legit, 1 = Fraud

# Train Model
model = RandomForestClassifier(n_estimators=50)
model.fit(X_train, y_train)

# Save Model
joblib.dump(model, "fraud_model.pkl")

def analyze_transaction(tx):
    features = np.array([[tx["amount"], tx["fee"], tx["coinjoin"]]])
    risk_score = model.predict_proba(features)[0][1] * 100
    return risk_score

### CODE REVIEW

**Comment by Horlabrainmoore** [comment]
> [!WARNING]
>  public_api.py
from flask import Flask, request, jsonify

app = Flask(__name__)
fraudulent_transactions = []

@app.route('/report', methods=['POST'])
def report_fraud():
    tx_data = request.json
    fraudulent_transactions.append(tx_data)
    return jsonify({"status": "Transaction reported!"})

@app.route('/get-reports', methods=['GET'])
def get_reports():
    return jsonify(fraudulent_transactions)

if __name__ == "__main__":
    app.run(debug=True, host="0.0.0.0", port=8000)
